web/react/components/toggle_modal_button.jsx

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+export default class ModalToggleButton extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.show = this.show.bind(this);
+        this.hide = this.hide.bind(this);
+
+        this.state = {
+            show: false
+        };
+    }
+
+    show() {
+        this.setState({show: true});
+    }
+
+    hide() {
+        this.setState({show: false});
+    }
+
+    render() {
+        const {children, dialogType, dialogProps, ...props} = this.props;
+
+        // this assumes that all modals will have a show property and an onHide event
+        const dialog = React.createElement(this.props.dialogType, Object.assign({}, dialogProps, {
+            show: this.state.show,
+            onHide: () => {
+                this.hide();
+
+                if (dialogProps.onHide) {
+                    dialogProps.onHide();
+                }
+            }
+        }));
+
+        // nesting the dialog in the anchor tag looks like it shouldn't work, but it does due to how react-bootstrap
```

Comment:
Is there something you wanted changed here? I added the comment since I actually expected this to not work initially and figured other people might end up coming across it and getting confused.